Orleans/src/Microsoft.Orleans.Host.Abstractions/AzureStorageProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    // this would go into the azure storage providers DLL
+    public static class AzureStorageProviderExtensions
```

Comment:
Yup, as I mentioned in the description, I just put everything in the same project for now and didn't even consider the namespace, just showing what the programing Api would look like (mainly startup class) . Also, some of these new files mention that they should go in different Dlls that you only pull in if you use them